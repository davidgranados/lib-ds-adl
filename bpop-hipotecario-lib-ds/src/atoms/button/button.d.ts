import { DefaultButtonProps } from "../../types/button";
import { ButtonBaseProps } from "../button-base";
import { HTMLPropsFor } from "../../types";

export type ButtonSize = "xs" | "sm" | "md" | "lg";

export interface ButtonProps extends DefaultButtonProps, ButtonBaseProps, HTMLPropsFor<"button"> {
  size?: ButtonSize;
}
