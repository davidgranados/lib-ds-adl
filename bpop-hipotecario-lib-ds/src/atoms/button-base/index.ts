export { default as ButtonBase } from "./button-base";

export type { ButtonBaseProps, ButtonFontSize } from "./button-base.d";
