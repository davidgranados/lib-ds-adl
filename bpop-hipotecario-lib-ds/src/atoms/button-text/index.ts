export { default as ButtonText } from "./button-text";

export { validSizes, validVariants } from "./helpers";
