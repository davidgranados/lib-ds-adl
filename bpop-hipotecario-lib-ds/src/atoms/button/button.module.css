.default {
  border: solid var(--border-width-button) var(--color-base-transparent);
  border-radius: var(--border-radius-button);
}

.primary {
  background-color: var(--background-color-button-primary-enabled);
  color: var(--font-color-button-primary-enabled);
}

.primary--focus {
  border: solid var(--border-width-button) var(--border-color-button-primary);
  background-color: var(--background-color-button-primary-focus);
}

.primary--hover {
  background-color: var(--background-color-button-primary-hover);
  color: var(--font-color-button-primary-hover);
}

.primary--active {
  background-color: var(--background-color-button-primary-active);
}

.primary:disabled {
  background-color: var(--background-color-button-primary-disabled);
  color: var(--font-color-button-primary-disabled);
}

.secondary {
  background-color: var(--background-color-button-secondary-enabled);
  color: var(--font-color-button-secondary-enabled);
}

.secondary--focus {
  border: solid var(--border-width-button) var(--border-color-button-secondary);
  background-color: var(--background-color-button-secondary-focus);
}

.secondary--hover {
  background-color: var(--background-color-button-secondary-hover);
  color: var(--font-color-button-secondary-hover);
}

.secondary--active {
  background-color: var(--background-color-button-secondary-active);
}

.secondary:disabled {
  background-color: var(--background-color-button-secondary-disabled);
  color: var(--font-color-button-secondary-disabled);
}

.size-xs {
  height: 32px;
  padding: 8px 16px;
}

.size-sm {
  height: 40px;
  padding: 8px 24px;
}

.size-md {
  height: 48px;
  padding: 12px 32px;
}

.size-lg {
  height: 56px;
  padding: 16px 32px;
}
