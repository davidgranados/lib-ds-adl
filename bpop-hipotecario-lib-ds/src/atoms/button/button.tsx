import React from "react";
import classNames from "classnames";

import { ButtonProps } from "../../types";

import buttonStyles from "./button.module.css";

export const Button: React.FC<ButtonProps> = ({
  label,
  type = "button",
  variant = "primary",
  size = "md",
  ...props
}) => {
  const buttonClassName = classNames(buttonStyles["default"], buttonStyles[variant], buttonStyles[`size-${size}`]);
  return (
    <button type={type} className={buttonClassName} {...props}>
      {label}
    </button>
  );
};
