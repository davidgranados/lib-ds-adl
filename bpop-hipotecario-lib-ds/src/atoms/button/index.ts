export { default as Button } from "./button";

export { validSizes, validVariants } from "./helpers";
