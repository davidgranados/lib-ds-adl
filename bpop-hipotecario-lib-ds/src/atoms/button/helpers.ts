import { ButtonVariant } from "../../types";
import { ButtonSize } from "./index";
import { ButtonFontSize } from "../button-base";

export const validSizes: ButtonSize[] = ["xs", "sm", "md", "lg"];
export const validVariants: ButtonVariant[] = ["primary", "secondary"];
export const fontSizeMapping: Record<ButtonSize, ButtonFontSize> = {
  xs: "sm",
  sm: "md",
  md: "md",
  lg: "lg",
};
