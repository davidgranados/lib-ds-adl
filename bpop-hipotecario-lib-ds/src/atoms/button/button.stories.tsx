/* istanbul ignore file */
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import Button from "./button";
import { validSizes, validVariants } from "./helpers";

export default {
  title: "Atoms/Button",
  component: Button,
  args: {
    variant: "primary",
    disabled: false,
    text: "Button",
  },
  argTypes: {
    variant: { options: validVariants, control: "select" },
    size: { options: validSizes, control: "select" },
    disabled: { control: "boolean" },
    onFocus: { action: "focused" },
    onBlur: { action: "blurred" },
    onClick: { action: "clicked" },
    onMouseDown: { action: "mouse down" },
    onMouseUp: { action: "mouse up" },
    onMouseEnter: { action: "mouse enter" },
    onMouseLeave: { action: "mouse leave" },
    onTouchStart: { action: "touch start" },
    onTouchEnd: { action: "touch end" },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Default = Template.bind({});
