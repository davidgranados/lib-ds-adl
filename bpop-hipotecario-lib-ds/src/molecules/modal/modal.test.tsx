import React from "react";
import { mount, shallow } from "enzyme";

import { Modal } from ".";
import { Button } from "../../atoms/button";

describe("<Modal />", () => {
  const component: JSX.Element = <Modal title="title" buttonText="ok" isDisabledButton={false} show={true} />;

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("Should match snapshot", () => {
    expect(mount(component)).toMatchSnapshot();
  });

  it("should toggle when click in button", () => {
    let show = false;
    const wrapper = shallow(component);
    wrapper.setProps({ show: show, onToggle: (s: boolean) => (show = s) });
    wrapper.find(Button).simulate("click");
    expect(show).toBe(true);
  });

  it("should toggle when click in overflow", () => {
    let show = false;
    const wrapper = shallow(component);
    wrapper.setProps({ show: show, onToggle: (s: boolean) => (show = s) });
    wrapper.find(".overflow").simulate("click");
    expect(show).toBe(true);
  });

  it("should toggle when click in icon close", () => {
    let show = false;
    const wrapper = mount(component);
    wrapper.setProps({ show: show, onToggle: (s: boolean) => (show = s) });
    wrapper.find('[data-testid="modal-icon-close"]').simulate("click");
    expect(show).toBe(true);
  });
});
