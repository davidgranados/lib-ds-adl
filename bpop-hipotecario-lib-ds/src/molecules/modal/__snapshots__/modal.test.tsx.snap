// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should match snapshot 1`] = `
<div
  className="wrapper show false"
>
  <div
    className="overflow"
    onClick={[Function]}
  />
  <div
    className="modal"
  >
    <div
      className="header"
    >
      <div
        className="close"
      >
        <span
          className="icon icon-close"
          onClick={[Function]}
        />
      </div>
      <div
        className="title"
      >
        <h6
          className="default h6 weight-bold carbon-900 line-height-h6"
        >
          title
        </h6>
      </div>
    </div>
    <div
      className="content"
    />
    <div
      className="footer"
    >
      <button
        className="default font-size-lg default primary size-lg"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        ok
      </button>
    </div>
  </div>
</div>
`;
